test/distributed/test_pg_wrapper.py

```diff
@@ -265,27 +265,72 @@ def test_collective_hang(self):
         def test_collectives_op_mismatch_debug_mode(self):
             pg = self._create_wrapper_pg(with_new_group=True)
             self._test_collectives_op_mismatch(pg, use_cuda=True)
+            self._test_nccl_only_op_mismatch(pg)
 
         @requires_nccl()
         @skip_if_lt_x_gpu(2)
         @with_dist_debug_levels(levels=["OFF"])
         def test_collectives_op_mismatch(self):
             pg = self._create_wrapper_pg(with_new_group=False)
             self._test_collectives_op_mismatch(pg, use_cuda=True)
+            self._test_nccl_only_op_mismatch(pg)
 
         @requires_nccl()
         @skip_if_lt_x_gpu(2)
         @with_dist_debug_levels(levels=["DETAIL"])
         def test_collective_shape_mismatch_debug_mode(self):
             pg = self._create_wrapper_pg(with_new_group=True)
             self._test_collective_shape_mismatch(pg, use_cuda=True)
+            self._test_nccl_only_shape_mismatch(pg)
 
         @requires_nccl()
         @skip_if_lt_x_gpu(2)
         @with_dist_debug_levels(levels=["OFF"])
         def test_collective_shape_mismatch(self):
             pg = self._create_wrapper_pg(with_new_group=False)
             self._test_collective_shape_mismatch(pg, use_cuda=True)
+            self._test_nccl_only_shape_mismatch(pg)
+
+        def _test_nccl_only_op_mismatch(self, wrapper_pg):
+            device = f"cuda:{self.rank}"
+            with self.assertRaisesRegex(RuntimeError, ".*") as cm:
+                output = torch.zeros(4 + self.rank, device=device)
+                input = torch.ones(4 * self.world_size, device=device)
+                if self.rank == 0:
+                    wrapper_pg._allgather_base(output, input).wait()
+                else:
+                    wrapper_pg._reduce_scatter_base(output, input).wait()
+            self._validate_error(
+                exception=cm.exception,
+                op_type="ALLGATHER_BASE" if self.rank == 0 else"REDUCE_SCATTER_BASE",
```

Comment:
strange, doesn't look like there is a space after `else` before `REDUCE_SCATTER_BASE`, wonder why python is not complaining

Reply:
I believe we don't lint the test suite.